prov/gni/src/gnix_mbox_allocator.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2015 Los Alamos National Security, LLC. All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include "gnix_mbox_allocator.h"
+
+int mbox_allocator_create(struct gnix_nic *nic, gni_cq_handle_t cq_handle,
+			  enum gnix_page_size page_size, size_t mbox_size,
+			  size_t mpmmap,
+			  struct gnix_mbox_alloc_handle *alloc_handle)
+{
+	return 0;
+}
+
+int mbox_alloc(struct gnix_mbox_alloc_handle *alloc_handle,
+	       struct gnix_mbox **ptr)
+{
+	return 0;
+}
+
+int mbox_free(struct gnix_mbox *ptr)
```

Comment:
Actually that requires keeping track of mailbox <-> allocator at a layer above this. Do we just want to shove a pointer to the allocator inside of each mailbox? It loses 8 bytes per mailbox, but is worth it to do that rather than rely on book keeping at the layer above? 

@hppritcha @ztiffany